### ISSUE TITLE
DPP error on large problem

### ISSUE THREAD

**Issue by bmeyers** [cvxpy/cvxpy#1622]
**Describe the bug**
When canonicalizing a largish QP (length 1e5), putting the problem in DPP form causes kernel death. This is likely due to the number of parameters involved. 

**To Reproduce**
```
# imports
import numpy as np
import cvxpy as cvx

# make problem data
T = int(1e5)
weight = 0.5
rho = 1.5
np.random.seed(42)
pwl = np.r_[
    np.linspace(0, 3, num=T//4),
    np.linspace(3, 1.5, num=T//6),
    np.linspace(1.5, -3, num=T//6),
    np.linspace(-3, -2.5, num=T//4),
    np.linspace(-2.5, 0, num=T//6+2)
]
y = pwl + 0.2 * np.random.randn(T)
use_set = np.random.uniform(size=T) >= .2
y[~use_set] = np.nan


# non-DPP compliant form
x = cvx.Variable(len(y))
Mv = cvx.Parameter(np.sum(use_set), value=y[use_set], name='Mv')
w = cvx.Parameter(value=weight, name='weight', nonneg=True)
r = cvx.Parameter(value=rho, name='rho', nonneg=True)
objective = cvx.Minimize(
    w * cvx.norm1(cvx.diff(x, k=2)) + r / 2 * cvx.sum_squares(
        x[use_set] - Mv
    )
)
problem = cvx.Problem(objective)
# this works but gives the non-DPP warning message
problem.solve()

# DPP compliant form
x = cvx.Variable(len(y))
Mv = cvx.Parameter(np.sum(use_set), value=y[use_set], name='Mv')
w = cvx.Parameter(value=weight, name='weight', nonneg=True)
r = rho
objective = cvx.Minimize(
    w * cvx.norm1(cvx.diff(x, k=2)) + r / 2 * cvx.sum_squares(
        x[use_set] - Mv
    )
)
problem = cvx.Problem(objective)
# this causes my kernel to die
problem.solve()
```

**Expected behavior**
It would be ideal if it were possible to solve this problem DPP form. But, given this limitation, there should be a way to turn off DPP canonicalization, even for DPP compliant problems. The `enforce_dpp` kwarg allows the user to do the opposite, to enforce that a problem must be DPP, but there's no apparent way to stop the DPP canonicalization, except by re-configuring the problem to be non-compliant. Additionally, if it is known that the process fails for problems or parameters above a certain size, some feedback or warning to the user would be helpful. Something like: "Caution: problems with more than XXX parameters can cause memory issues when creating DPP canonicalizations. If you experience issues, turn off DPP for this problem by setting `some_new_kwarg=False`".

**Output**
Error message generated by the Jupyter Notebook that was running the code pasted above:

<img width="775" alt="Screen Shot 2022-01-28 at 12 56 05 PM" src="https://user-images.githubusercontent.com/1463184/151626716-ca7a676d-77e3-45ef-889f-ee341b652f50.png">

**Version**
 - OS: macOS 10.15.7
 - CVXPY Version: 1.1.18

**Additional context**
The problem is a QP that comes up as in the context of proximal operators for [signal decomposition](https://github.com/bmeyers/optimal-signal-demixing/).

**Comment by SteveDiamond** [comment]
@bmeyers sorry for the confusing behavior. 100% agree that DPP should be easy to disable and should give a warning when the dimensions of the parameters are large (maybe > 1e4 all added together?).

**Comment by bmeyers** [comment]
That sounds good to me, thank you!

**Comment by rileyjmurray** [comment]
Resolved by #1669. Thanks for reporting, @bmeyers!
